lib/beaker/host.rb

```diff
@@ -218,16 +231,22 @@ def is_x86_64?
     #  host.add_env_var('PATH', '/usr/bin:PATH')
     def add_env_var key, val
       key = key.to_s.upcase
-      escaped_val = Regexp.escape(val).gsub('/', '\/').gsub(';', '\;')
-      env_file = self[:ssh_env_file]
-      #see if the key/value pair already exists
-      if exec(Beaker::Command.new("grep -e #{key}=.*#{escaped_val} #{env_file}"), :acceptable_exit_codes => (0..255) ).exit_code == 0
-        return #nothing to do here, key value pair already exists
-      #see if the key already exists
-      elsif exec(Beaker::Command.new("grep #{key} #{env_file}"), :acceptable_exit_codes => (0..255) ).exit_code == 0
-        exec(Beaker::SedCommand.new(self['HOSTS'][name]['platform'], "s/#{key}=/#{key}=#{escaped_val}:/", env_file))
-      else
-        exec(Beaker::Command.new("echo \"#{key}=#{val}\" >> #{env_file}"))
+      if self.is_cygwin?
+        env_file = self[:ssh_env_file]
+        escaped_val = Regexp.escape(val).gsub('/', '\/').gsub(';', '\;')
+        #see if the key/value pair already exists
+        if exec(Beaker::Command.new("grep -e #{key}=.*#{escaped_val} #{env_file}"), :acceptable_exit_codes => (0..255) ).exit_code == 0
+          return #nothing to do here, key value pair already exists
+        #see if the key already exists
+        elsif exec(Beaker::Command.new("grep #{key} #{env_file}"), :acceptable_exit_codes => (0..255) ).exit_code == 0
+          exec(Beaker::SedCommand.new(self['HOSTS'][name]['platform'], "s/#{key}=/#{key}=#{escaped_val}:/", env_file))
+        else
+          exec(Beaker::Command.new("echo \"#{key}=#{val}\" >> #{env_file}"))
+        end
+      else #powershell windows
+        #see if the key/value pair already exists
+        exec(Beaker::Command.new("set #{key}=%#{key}%;#{val}"))
+        exec(Beaker::Command.new("setx #{key} %#{key}%;#{val}"))
```

Comment:
This will need testing because I had the same thing in my own branch and found that the setx command wasn't always working and required the ```host.close``` hack you have above in order to be available.